### PR TITLE
fix: decide travel arrival by elapsed time, not integer truncation

### PR DESCRIPTION
**Pull request by clintongormley**
## Summary

`TravelCalculator._calculate_position` truncated the intermediate position with `int()`, and `position_reached()` compares that integer with the target. For a decreasing move the tracker equalled the target as soon as the true value dropped below `target + 1`; an increasing move only arrived when the time check returned the target. Consequences, verified in review against a real switch-mode cover:

- A 50→30 move on a 60 s cover "arrived" at 95 % and cut the relay early; 30→50 waited for the clock.
- A one-step close (51→50) arrived on its **first tick**; a two-step close at ~58 %. Short closes barely moved the motor.
- Repeated partial closes drifted "more open" than tracked, and the `is_traveling()` window that reversal logic depends on was narrowed. Overhead calibration under-drove every close-direction step.

The fix rounds the intermediate position and holds it one step short of the target until the existing elapsed-time check declares arrival, so both directions arrive at exactly 100 % of the travel time. Value-level rather than a separate predicate because two external callers compare `current_position()` against a target directly (`cover_base` endpoint settle, the calibration loop).

Visible consequence: a cover reads `closed`/`open` only when its travel time has elapsed, and a stop in the final half-step leaves it one step short rather than at the endpoint (physically it is at ~0.5 %; the old value lied the other way).

## Tests

- New calculator tests: long, one-step, two-step and endpoint moves in both directions (mid-move not reached, reached at full time), symmetric rounding. They fail on `main` (6 of 19).
- No pre-existing test changed: nothing in the suite depended on the truncation.
- Review ran a 20 000-move fuzz (never early, never more than one step from exact, monotonic, in range, arrival at 100 %) with 0 violations at head and ~93 000 early arrivals at base, plus real-HA runs of short closes showing the relay now held for the full step time.

## Also

Removes two stale duplicate lines from the 4.12.0 changelog (the pre-review wording of #265's two entries, which a rebase replayed as additions next to their final wording).

### CODE REVIEW

File: custom_components/cover_time_based/travel_calculator.py [hunk 1]
```diff
@@ -259,14 +265,21 @@ def position_reached_or_exceeded(relative_position: int) -> bool:
         )
         if remaining_travel_time <= 0:
             return self._travel_to_position
-        if time.time() > self._last_known_position_timestamp + remaining_travel_time:
+        now = time.time()
+        if now > self._last_known_position_timestamp + remaining_travel_time:
             return self._travel_to_position
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply custom_components/cover_time_based/travel_calculator.py:270]: Good catch, and it was a regression of this very change: before the hold, truncating an exact-boundary value already gave the target, so the strict `>` was harmless. Fixed in 37bd98b: the comparison is now `>=`, and a new parametrised test pins arrival at exactly the boundary in both directions (it failed with `31 == 30` / `49 == 50` before the fix). The review fuzz sampled just past the boundary and stepped over this edge; it now asserts the exact boundary too.
